examples/QuickJournalSync/Services/RealmService.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using MongoDB.Bson;
+using QuickJournalSync.Models;
+using Realms;
+using Realms.Sync;
+using Realms.Sync.ErrorHandling;
+using Realms.Sync.Exceptions;
+
+namespace QuickJournalSync.Services
+{
+    public static class RealmService
+    {
+        private static readonly string _appId = "";
+
+        private static object _mainRealmLock = new();
+
+        private static bool _serviceInitialised;
+
+        private static Realms.Sync.App? _app;
+
+        private static Realm? _mainThreadRealm;
+
+        public static User? CurrentUser => _app?.CurrentUser;
+
+        public static event EventHandler<ConnectionState>? SyncConnectionStateChanged;
+
+        public static void Init()
+        {
+            if (_serviceInitialised)
+            {
+                return;
+            }
+
+            if (string.IsNullOrEmpty(_appId))
+            {
+                throw new Exception("Remember to add your appId!");
+            }
+
+            var appConfiguration = new AppConfiguration(_appId);
+
+            _app = Realms.Sync.App.Create(appConfiguration);
+
+            _serviceInitialised = true;
+        }
+
+        public static Realm GetMainThreadRealm()
+        {
+            lock (_mainRealmLock)
+            {
+                if (_mainThreadRealm is null)
+                {
+                    var mainThreadConfig = GetRealmConfig(sessionErrorCallback: HandleSessionErrorCallback,
+                        clientResetHandler: MakeClientResetHandler());
+                    _mainThreadRealm = Realm.GetInstance(mainThreadConfig);
+                    _mainThreadRealm.SyncSession.PropertyChanged += HandleSyncSessionPropertyChanged;
+                }
+            }
+
+            return _mainThreadRealm;
```

Comment:
Should we have some guard that asserts that we're indeed on the main thread? Doesn't seem like anything is guaranteeing this here.

Reply:
We can do that